### ISSUE TITLE
SDWebImagePrefetcher completionBlock bug

### ISSUE THREAD

**Issue by threeknife** [rs/SDWebImage#1456]
```
(void)startPrefetchingAtIndex:(NSUInteger)index {
    if (index >= self.prefetchURLs.count) return;
    self.requestedCount++;
    [self.manager downloadImageWithURL:self.prefetchURLs[index] options:self.options progress:nil completed:^(UIImage *image, NSError *error, SDImageCacheType cacheType, BOOL finished, NSURL *imageURL) {
        if (!finished) return;
        self.finishedCount++;

        if (image) {
            if (self.progressBlock) {
                self.progressBlock(self.finishedCount,[self.prefetchURLs count]);
            }
        }
        else {
            if (self.progressBlock) {
                self.progressBlock(self.finishedCount,[self.prefetchURLs count]);
            }
            // Add last failed
            self.skippedCount++;
        }
        if ([self.delegate respondsToSelector:@selector(imagePrefetcher:didPrefetchURL:finishedCount:totalCount:)]) {
            [self.delegate imagePrefetcher:self
                            didPrefetchURL:self.prefetchURLs[index]
                             finishedCount:self.finishedCount
                                totalCount:self.prefetchURLs.count
             ];
        }
        else if (self.finishedCount == self.requestedCount) {
            [self reportStatus];
            if (self.completionBlock) {
                self.completionBlock(self.finishedCount, self.skippedCount);
               self.completionBlock = nil;
            }
            self.progressBlock = nil;
        }
    }];
}
```

I found that the completionblock will be excute when  self.finishedCount == self.requestedCount == 1,
I think this a critical bug and that I can't user  SDWebImagePrefetcher anymore.

**Comment by mythodeia** [comment]
@threeknife are you using the latest version? there were some fixes concerning this 10 days ago i think

**Comment by rushk2112** [comment]
As of SDWebImage (3.7.5) this appears to be an issue still. I fed a list of 36 images to the prefetched but they do not report 36 done, nor does the process seem to update. I am try to isolate this to a possible threading issue in my code but the prefetched does not report the number of images in the original array.

**Comment by bpoplauschi** [comment]
This issue should have been fixed in 3.7.5 a8fca6584feabd625abdcbc0da921cbfc689fa3e. Please reopen if it still exists
